aws/src/main/scala/com/gu/contentapi/client/IAMSigner.scala

```diff
@@ -0,0 +1,44 @@
+package com.gu.contentapi.client
+
+import com.amazonaws.DefaultRequest
+import com.amazonaws.auth.{AWS4Signer, AWSCredentials}
+import com.amazonaws.http.HttpMethodName
+import collection.JavaConverters._
+
+object IAMSigner {
+  /**
+    * For api-gateway authorization.
+    * Returns the given set of headers, updated to include AWS sig4v signed headers based on the request and the credentials
+    */
+  def addIAMHeaders(headers: Map[String, String], url: String, credentials: AWSCredentials, awsRegion: String): Map[String, String] = {
```

Comment:
I don't know anything about api gateway, but aren't `credentials` and `awsRegion` fixed once and for all?

Reply:
sadly not - https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/AWSCredentials.html

Reply:
I may have misunderstood your question, were you asking if the credentials object is already linked to the region?

Reply:
No I was wondering if the credentials and the region were going to change from one call to the other in the same "session"?

Reply:
Yes they'll be the same throughout runtime

Reply:
I was going to say it may be worth turning that object into a class then. Admittedly the `AWS4Signer` constructor is not doing too much and delays everything until the `sign` call.

Reply:
I agree actually, will make it a class